api/urls.py

```diff
@@ -15,48 +22,97 @@ def register_route(prefix, viewset):
-# register_creative_work_route(r'thesis')
-# register_creative_work_route(r'workingpaper')
-
-# # creative work attributes
-# register_route(r'tag', views.TagViewSet)
-# register_route(r'subject', views.SubjectViewSet)
-# register_route(r'extra', views.ExtraDataViewSet)
-
-# # agent routes
-# register_route(r'person', views.PersonViewSet)
-# register_route(r'organization', views.OrganizationViewSet)
-# register_route(r'institution', views.InstitutionViewSet)
-
-# # identifier routes
-# register_route(r'agentidentifier', views.AgentIdentifierViewSet)
-# register_route(r'workidentifier', views.WorkIdentifierViewSet)
-
-# # relation routes
-# register_route(r'agentrelation', views.AgentRelationViewSet)
-# register_route(r'workrelation', views.WorkRelationViewSet)
-# register_route(r'contribution', views.ContributionViewSet)
-# register_route(r'award', views.AwardViewSet)
+class BaseShareSerializer(serializers.ModelSerializer):
+
+    def __init__(self, *args, **kwargs):
+        # super hates my additional kwargs
+        sparse = kwargs.pop('sparse', False)
+        version_serializer = kwargs.pop('version_serializer', False)
+        super(BaseShareSerializer, self).__init__(*args, **kwargs)
+
+        if sparse:
+            # clear the fields if they asked for sparse
+            self.fields.clear()
+        else:
+            # remove hidden fields
+            excluded_fields = ['change', 'uuid', 'sources']
+            for field_name in tuple(self.fields.keys()):
+                if 'version' in field_name or field_name in excluded_fields:
+                    self.fields.pop(field_name)
+
+            if not version_serializer:
+                # add links to related objects
+                self.fields.update({
+                    'links': fields.LinksField(links=self.Meta.links, source='*')
+                })
+
+        # version specific fields
+        if version_serializer:
+            self.fields.update({
+                'action': serializers.CharField(max_length=10),
+                'persistent_id': serializers.IntegerField()
+            })
+
+        # add fields with improper names
+        self.fields.update({
+            'type': fields.TypeField(),
+        })
+
+    class Meta:
+        links = ('versions', 'changes', 'rawdata')
+
+    # http://stackoverflow.com/questions/27015931/remove-null-fields-from-django-rest-framework-response
+    def to_representation(self, instance):
+        def not_none(value):
+            return value is not None
+
+        ret = super(BaseShareSerializer, self).to_representation(instance)
+        ret = OrderedDict(list(filter(lambda x: not_none(x[1]), ret.items())))
+        return ret
+
+
+class EndpointGenerator:
+
+    def __init__(self):
+        subclasses = models.ShareObject.__subclasses__()
+
+        generated_endpoints = []
+        for subclass in subclasses:
+            if not (subclass._meta.proxied_children and subclass == subclass._meta.concrete_model):
```

Comment:
When testing for the equality of two classes `is` is preferred as classes (and primitives) are always the same object.

```python
In [1]: dict == dict
Out[1]: True

In [2]: dict is dict
Out[2]: True

In [3]: dict is type({})
Out[3]: True

In [4]: type({}) is type({})
Out[4]: True

In [5]: {} == {}
Out[5]: True

In [6]: {} is {}
Out[6]: False
```